tools/cldr-code/src/main/java/org/unicode/cldr/tool/GeneratePersonNameTestData.java

```diff
@@ -0,0 +1,170 @@
+package org.unicode.cldr.tool;
```

Comment:
I got lost in this file.  What I think I eventually pieced together is that this program generates test data in the above format from the sample-name data in CLDR.  Is that correct?

Equally important, is that what we want to do?  I guess it's a good place to start, but the CLDR sample-name data seems like it might be a fairly imperfect vehicle for testing all the various features of our engine, and in particular to test that the ICU version of the code does the same thing as the CLDR version of the code.  Then again, maybe I'm wrong about this.  I _do_ think it's a perfectly reasonable place to start, but that we might need to augment it with some hand-created test cases for particular features.  Or is that what you're planning to do?

Reply:
I think testing all the files that have person name data, with the 8 sample names and all combinations of parameters should be a good start. Maybe even overkill, but we can cut them down if the test is too slow.

We can also augment them with other tests, like checking individual patterns.

Reply:
I added a preliminary chunk of code that tests against the ICU implementation. I think if we  use the native and foreign names for each locale, and pass all the parameters combinations, then we are in really good shape. If it turns out that test takes too long, then we could look at limiting it.

Reply:
That sounds reasonable.